### PR TITLE
Enable nullable checks for markdown, TOC markdown, git

### PR DESCRIPTION
**Pull request by yufeih**
#5515 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5534)

### CODE REVIEW

**Comment by mingwli** [comment]
> ⚠[DevSandbox](https://github.com/OPS-E2E-PPE/DevSandbox)(00:00:20.3834188, 0 more diff)

😂😂😂

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 1]
```diff
@@ -201,15 +200,15 @@ private string GetLink(SourceInfo<string> href)
...
             var (error, monikers) = _monikerProvider.GetZoneLevelMonikers(((Document)InclusionContext.RootFile).FilePath, monikerRange);
             status.Errors.AddIfNotNull(error);
             return monikers;
         }
 
-        private sealed class Status
+        private class Status
         {
-            public List<Error> Errors;
+            public List<Error> Errors = new List<Error>();
```
Comment (live1206) [inline src/docfx/lib/markdown/MarkdownEngine.cs]: should be readonly

**Pull request merged**